src/lib.rs

```diff
@@ -224,13 +233,15 @@ where
                 archive_reader,
             )?,
             Mode::WriteDisk => {
-                let writer_flags = (ffi::ARCHIVE_EXTRACT_TIME
+                let mut writer_flags = (ffi::ARCHIVE_EXTRACT_TIME
                     | ffi::ARCHIVE_EXTRACT_PERM
                     | ffi::ARCHIVE_EXTRACT_ACL
                     | ffi::ARCHIVE_EXTRACT_FFLAGS
-                    | ffi::ARCHIVE_EXTRACT_OWNER
                     | ffi::ARCHIVE_EXTRACT_XATTR) as i32;
 
+                if let Options::PreserveOwnership = options {
+                    writer_flags |= ffi::ARCHIVE_EXTRACT_OWNER as i32;
```

Comment:
I think it's better to let the `writer_flag` be a u32 identify and only cast it when used as an argument to avoid the double casting. I'm leaving a suggestion blow this.